src/foam/dao/EasyDAO.js

```diff
@@ -131,6 +132,13 @@ if ( head instanceof foam.dao.MDAO ) {
   }
 }
 
+if ( getFixedSizeDAO() &&
+     ( head instanceof foam.dao.MDAO || head instanceof foam.dao.NullDAO ) &&
```

Comment:
What's the idea behind these checks? If `fixedSizeDAO` is set to true and we're silently not adding that decorator in certain situations then it seems like it would be very easy to introduce unintended bugs where we expect the decorator to be there but it isn't.